ocaml/xapi/dbsync_master.ml

```diff
@@ -107,90 +107,23 @@ let release_locks ~__context =
       warn "failed to destroy redundant tools SR %s: %s"
         (Ref.string_of self) (Printexc.to_string e) in
-  let sr =
-    let srs = Db.SR.get_all ~__context in
-    let tools_srs =
-      List.filter (fun self -> Db.SR.get_is_tools_sr ~__context ~self) srs in
-    let old_srs =
-      List.filter (fun self ->
-          let other_config = Db.SR.get_other_config ~__context ~self in
-          (Db.SR.get_is_tools_sr ~__context ~self = false) &&
-          ((List.mem_assoc Xapi_globs.tools_sr_tag other_config) ||
-           (List.mem_assoc Xapi_globs.xensource_internal other_config))
-        ) srs in
-    match tools_srs with
-    | sr :: others ->
-      (* Let there be only one Tools SR *)
-      List.iter destroy others;
-      List.iter destroy old_srs;
-      sr
-    | [] ->
-      (* First check if there is an SR with the old tags on it, which needs upgrading (set is_tools_sr). *)
-      (* We cannot do this in xapi_db_upgrade, because that runs later. *)
-      match old_srs with
-      | sr :: others ->
-        Db.SR.set_is_tools_sr ~__context ~self:sr ~value:true;
-        List.iter destroy others; (* destroy bogus Tool SRs CA-300103 *)
-        sr
-      | [] ->
-        create_magic_sr name_label name_description other_config
-  in
-  (* Ensure fields are up-to-date *)
-  Db.SR.set_name_label ~__context ~self:sr ~value:name_label;
-  Db.SR.set_name_description ~__context ~self:sr ~value:name_description;
-  let other_config =
-    (* Keep any existing keys/value pair besides the required ones *)
-    let oc = Db.SR.get_other_config ~__context ~self:sr in
-    let keys = List.map fst other_config in
-    let extra = List.filter (fun (k, _) -> not (List.mem k keys)) oc in
-    extra @ other_config
-  in
-  Db.SR.set_other_config ~__context ~self:sr ~value:other_config;
-  List.iter
-    (fun self ->
-       Db.PBD.set_device_config ~__context ~self
-         ~value:Xapi_globs.tools_sr_pbd_device_config)
-    (Db.SR.get_PBDs ~__context ~self:sr)
-
-let create_tools_sr_noexn __context =
-  let name_label = Xapi_globs.tools_sr_name () in
-  let name_description = Xapi_globs.tools_sr_description () in
-  Helpers.call_api_functions ~__context (fun rpc session_id ->
-      let sr_introduce = Client.SR.introduce ~rpc ~session_id in
-      let maybe_create_pbd  = Create_storage.maybe_create_pbd rpc session_id in
-      Helpers.log_exn_continue "creating tools SR" (fun () ->
-          create_tools_sr __context name_label name_description sr_introduce maybe_create_pbd
-        ) ()
-    )
+  let srs = Db.SR.get_all ~__context in
+  let tools_srs =
+    List.filter (fun self -> Db.SR.get_is_tools_sr ~__context ~self) srs in
+  let old_srs =
+    List.filter (fun self ->
+        let other_config = Db.SR.get_other_config ~__context ~self in
+        (Db.SR.get_is_tools_sr ~__context ~self = false) &&
+        ((List.mem_assoc Xapi_globs.tools_sr_tag other_config) ||
+         (List.mem_assoc Xapi_globs.xensource_internal other_config))
```

Comment:
why does having a `xensource_internal` property mean it's an old sr?

Reply:
It's from the days before we had a proper `is_tools_sr` field, and even from before the days that we had a `tools_sr_tag` other_config key.

Reply:
this check has just been moved, I'd rather keep the logic the same in here unless there's a reason to change it.

Reply:
Indeed.